aws_lambda_powertools/utilities/parameters/ssm.py

```diff
@@ -0,0 +1,212 @@
+        ----------
+        path: str
+            Path to retrieve the parameters
+        decrypt: bool, optional
+            If the parameter values should be decrypted
+        recursive: bool, optional
+            If this should retrieve the parameter values recursively or not
+        sdk_options: dict, optional
+            Dictionary of options that will be passed to the get_parameters_by_path call
+        """
+
+        # Explicit arguments will take precedence over keyword arguments
+        sdk_options["Path"] = path
+        sdk_options["WithDecryption"] = decrypt
+        sdk_options["Recursive"] = recursive
+
+        response = self.client.get_parameters_by_path(**sdk_options)
+        parameters = response.get("Parameters", [])
+
+        # Keep retrieving parameters
+        while "NextToken" in response:
+            response = self.client.get_parameters_by_path(
+                Path=path, WithDecryption=decrypt, Recursive=recursive, NextToken=response["NextToken"]
+            )
+            parameters.extend(response.get("Parameters", []))
+
+        retval = {}
+        for parameter in parameters:
+
+            # Standardize the parameter name
+            # The parameter name returned by SSM will contained the full path.
+            # However, for readability, we should return only the part after
+            # the path.
+            name = parameter["Name"]
+            if name.startswith(path):
+                name = name[len(path) :]
+            name = name.lstrip("/")
+
+            retval[name] = parameter["Value"]
+
+        return retval
+
+
+def get_parameter(name: str, transform: Optional[str] = None, **sdk_options) -> Union[str, list, dict, bytes]:
+    """
+    Retrieve a parameter value from AWS Systems Manager (SSM) Parameter Store
+
+    Parameters
+    ----------
+    name: str
+        Name of the parameter
+    transform: str, optional
+        Transforms the content from a JSON object ('json') or base64 binary string ('binary')
+    sdk_options: dict, optional
+        Dictionary of options that will be passed to the get_secret_value call
+
+    Example
+    -------
+    **Retrieves a parameter value from Systems Manager Parameter Store**
+
+        >>> from aws_lambda_powertools.utilities.parameters import get_parameter
+        >>>
+        >>> get_parameter("/my/parameter")
+
+    **Retrieves a parameter value and decodes it using a Base64 decoder**
+
+        >>> from aws_lambda_powertools.utilities.parameters import get_parameter
+        >>>
+        >>> get_parameter("/my/parameter", transform='binary')
+    """
+
+    # Only create the provider if this function is called at least once
+    if "ssm" not in DEFAULT_PROVIDERS:
+        DEFAULT_PROVIDERS["ssm"] = SSMProvider()
+
+    return DEFAULT_PROVIDERS["ssm"].get(name, transform=transform)
+
+
+def get_parameters(
+    path: str, transform: Optional[str] = None, recursive: bool = False, decrypt: bool = False, **sdk_options
```

Comment:
I've had many customers complaining about the default value of `recursive` in the CLI/SDK.

Imho, for 99% of use cases customers want `recursive=True`, even though the default is false.

For example, if you have these parameters:
- /app/dev/db/collection
- /app/dev/db/host
- /app/dev/bucket

A developer would expect `get_parameters("/app/dev/")` to return three parameters, but it won't return anything. This is very confusing and frustrating.

You'd need to `get_parameters("/app/dev/", recursive=True)` to fetch everything.

That's why I made recursive=True the default value in https://github.com/alexcasalboni/ssm-cache-python